src/client/datascience/jupyter/interpreter/jupyterInterpreterDependencyService.ts

```diff
@@ -125,104 +125,108 @@ export class JupyterInterpreterDependencyService {
     @reportAction(ReportableAction.InstallingMissingDependencies)
     public async installMissingDependencies(
         interpreter: PythonEnvironment,
-        _error?: JupyterInstallError,
-        token?: CancellationToken
+        _error?: JupyterInstallError
     ): Promise<JupyterInterpreterDependencyResponse> {
-        // If we're dealing with a non-conda environment & pip isn't installed, we can't install anything.
-        // Hence prompt to install pip as well.
-        const pipInstalledInNonCondaEnvPromise =
-            interpreter.envType === EnvironmentType.Conda
-                ? Promise.resolve(undefined)
-                : this.installer.isInstalled(Product.pip, interpreter);
+        const tokenSource = new CancellationTokenSource();
+        try {
+            // If we're dealing with a non-conda environment & pip isn't installed, we can't install anything.
+            // Hence prompt to install pip as well.
+            const pipInstalledInNonCondaEnvPromise =
+                interpreter.envType === EnvironmentType.Conda
+                    ? Promise.resolve(undefined)
+                    : this.installer.isInstalled(Product.pip, interpreter);
 
-        const [missingProducts, pipInstalledInNonCondaEnv] = await Promise.all([
-            this.getDependenciesNotInstalled(interpreter, token),
-            pipInstalledInNonCondaEnvPromise
-        ]);
-        if (Cancellation.isCanceled(token)) {
-            return JupyterInterpreterDependencyResponse.cancel;
-        }
-        if (missingProducts.length === 0) {
-            return JupyterInterpreterDependencyResponse.ok;
-        }
+            const [missingProducts, pipInstalledInNonCondaEnv] = await Promise.all([
+                this.getDependenciesNotInstalled(interpreter, tokenSource.token),
+                pipInstalledInNonCondaEnvPromise
+            ]);
+            if (Cancellation.isCanceled(tokenSource.token)) {
```

Comment:
This doesn't seem necessary anymore, this cancellation is created a few lines above, hence there's no way this can get cancelled.

Reply:
Yeah it's not going to be canceled by anything